### PR TITLE
add doctrine phpcr-bundle

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
| Q             | A
| ------------- | ---
| License       | MIT

This one will add a reciepe for doctrine phpcr-bundle.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
@dbu you got some input here?

File: doctrine/phpcr-bundle/2.0/config/packages/doctrine_phpcr.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#    odm:
+#        auto_mapping: true
+#        auto_generate_proxy_classes: "%kernel.debug%"
+#    session:
+#        backend:
+#            type: doctrinedbal
+#            connection: default
+#        workspace: default
+#        username: '%env(PHPCR_SESSION_USERNAME)%'
+#        password: '%env(PHPCR_SESSION_PASSWORD)%'
```
Comment (fabpot) [inline doctrine/phpcr-bundle/2.0/config/packages/doctrine_phpcr.yaml]: We do not copy/paste the default configuration as it means syncing it whenever something change upstream. Instead, we only list the ones which needs to be changed by the end user. If there is none, then the file can be removed.

Comment (ElectricMaxxx) [reply doctrine/phpcr-bundle/2.0/config/packages/doctrine_phpcr.yaml]: Mh, ok sounds valid. But then i would have a question/feature request: is it possible to use env vars as default on configuration directly? I mean would `->defaultValue( '%env(PHPCR_SESSION_USERNAME)%')` work? Or shouldn't we have an extra `defaultEnvironmentValue('PHPCR_SESSION_USERNAME', optonalCast)` for that?

Comment (Nyholm) [reply doctrine/phpcr-bundle/2.0/config/packages/doctrine_phpcr.yaml]: The bundle should not handle environment variables. The bundle should only care about its config. 

When defining your config in `Configuration.php` you should use `->defaultValue('foobar')`. When the user is using the config in `config.yaml` they might want to use `%env(FOO)%`.

Comment (ElectricMaxxx) [reply doctrine/phpcr-bundle/2.0/config/packages/doctrine_phpcr.yaml]: So providing a configuration for both credential parts (username, password) set by envirionment variables here would be a good example?
Then i would remove the other configuration parts. But i think i would keep the decission for `doctrinedbal` and add a hint, that there is the place to switch to an other backend, and that dbal would need doctrine dbal configuration. Doing so we would have one working exampe then.

**Comment by dbu** [comment]
if i understand correctly, flex basically makes the doctrine/phpcr-bundle uninstallable on its own. it will automatically enable the bundle, but the bundle can't run without a configured transport. but for the transport, we need a decision between jackalope-jackrabbit and jackalope-doctrine-dbal. that decision is another composer dependency.

we could provide two recipies for phpcr-odm with jackalope-jackrabbit and with dbal. for that we would have to create a repository for each of those with a composer.json that requires the phpcr bundle and the right jackalope version. then the recipies for those can also provide meaningful defaults.

if there is an easier solution it would be great. so if anybody has one, please tell!

**Comment by Nyholm** [comment]
For the record: https://github.com/symfony-cmf/symfony-cmf/issues/259

**Comment by maxhelias** [comment]
Same as #181

**Comment by Nyholm** [comment]
Closing due inactivity. Feel free to comment or reopen. 

#eu-fossa

**Pull request closed without merging**